presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSphericalGeoFunctions.java

```diff
@@ -0,0 +1,148 @@
+        }
+    }
+
+    @Test
+    public void testToAndFromSphericalGeography()
+    {
+        // empty geometries
+        assertToSphericalGeographyAndToGeometry("POINT EMPTY");
+        assertToSphericalGeographyAndToGeometry("MULTIPOINT EMPTY");
+        assertToSphericalGeographyAndToGeometry("LINESTRING EMPTY");
+        assertToSphericalGeographyAndToGeometry("MULTILINESTRING EMPTY");
+        assertToSphericalGeographyAndToGeometry("POLYGON EMPTY");
+        assertToSphericalGeographyAndToGeometry("MULTIPOLYGON EMPTY");
+        assertToSphericalGeographyAndToGeometry("GEOMETRYCOLLECTION EMPTY");
+
+        // valid nonempty geometries
+        assertToSphericalGeographyAndToGeometry("POINT (-40.2 28.9)");
+        assertToSphericalGeographyAndToGeometry("MULTIPOINT ((-40.2 28.9), (-40.2 31.9))");
+        assertToSphericalGeographyAndToGeometry("LINESTRING (-40.2 28.9, -40.2 31.9, -37.2 31.9)");
+        assertToSphericalGeographyAndToGeometry("MULTILINESTRING ((-40.2 28.9, -40.2 31.9), (-40.2 31.9, -37.2 31.9))");
+        assertToSphericalGeographyAndToGeometry("POLYGON ((-40.2 28.9, -37.2 28.9, -37.2 31.9, -40.2 31.9, -40.2 28.9))");
+        assertToSphericalGeographyAndToGeometry("POLYGON ((-40.2 28.9, -37.2 28.9, -37.2 31.9, -40.2 31.9, -40.2 28.9), " +
+                "(-39.2 29.9, -39.2 30.9, -38.2 30.9, -38.2 29.9, -39.2 29.9))");
+        assertToSphericalGeographyAndToGeometry("MULTIPOLYGON (((-40.2 28.9, -37.2 28.9, -37.2 31.9, -40.2 31.9, -40.2 28.9)), " +
+                "((-39.2 29.9, -38.2 29.9, -38.2 30.9, -39.2 30.9, -39.2 29.9)))");
+        assertToSphericalGeographyAndToGeometry("GEOMETRYCOLLECTION (POINT (-40.2 28.9), LINESTRING (-40.2 28.9, -40.2 31.9, -37.2 31.9), " +
+                "POLYGON ((-40.2 28.9, -37.2 28.9, -37.2 31.9, -40.2 31.9, -40.2 28.9)))");
+
+        // geometries containing invalid latitude or longitude values
+        assertInvalidLongitude("POINT (-340.2 28.9)");
+        assertInvalidLatitude("MULTIPOINT ((-40.2 128.9), (-40.2 31.9))");
+        assertInvalidLongitude("LINESTRING (-40.2 28.9, -40.2 31.9, 237.2 31.9)");
+        assertInvalidLatitude("MULTILINESTRING ((-40.2 28.9, -40.2 31.9), (-40.2 131.9, -37.2 31.9))");
+        assertInvalidLongitude("POLYGON ((-40.2 28.9, -40.2 31.9, 237.2 31.9, -37.2 28.9, -40.2 28.9))");
+        assertInvalidLatitude("POLYGON ((-40.2 28.9, -40.2 31.9, -37.2 131.9, -37.2 28.9, -40.2 28.9), (-39.2 29.9, -39.2 30.9, -38.2 30.9, -38.2 29.9, -39.2 29.9))");
+        assertInvalidLongitude("MULTIPOLYGON (((-40.2 28.9, -40.2 31.9, -37.2 31.9, -37.2 28.9, -40.2 28.9)), " +
+                "((-39.2 29.9, -39.2 30.9, 238.2 30.9, -38.2 29.9, -39.2 29.9)))");
+        assertInvalidLatitude("GEOMETRYCOLLECTION (POINT (-40.2 28.9), LINESTRING (-40.2 28.9, -40.2 131.9, -37.2 31.9), " +
+                "POLYGON ((-40.2 28.9, -40.2 31.9, -37.2 31.9, -37.2 28.9, -40.2 28.9)))");
+    }
+
+    private void assertToSphericalGeographyAndToGeometry(String wkt)
```

Comment:
nit: perhaps, rename to assertToAndFromSphericalGeography to match testToAndFromSphericalGeography

Reply:
@gaoxm I think xxxToAndFromSphericalGeography names are clearer than xxxToSphericalGeographyAndToGeometry